src/opendr/perception/object_detection_2d/nanodet/nanodet_learner.py

```diff
@@ -333,11 +360,147 @@ def reset(self):
         """This method is not used in this implementation."""
         return NotImplementedError
 
-    def optimize(self):
-        """This method is not used in this implementation."""
-        return NotImplementedError
+    def _save_onnx(self, onnx_path, img=None, do_constant_folding=False, verbose=True):
+        if self.jit_model:
+            print("Warning: A jit model has already initialized, inference will run in onnx mode by default!!!")
+        if not self.predictor:
+            self.predictor = Predictor(self.cfg, self.model, device=self.device)
+
+        os.makedirs(onnx_path, exist_ok=True)
+        export_path = os.path.join(onnx_path, "nanodet_{}.onnx".format(self.cfg.check_point_name))
+
+        if self.dummy_input is None:
+            assert img is not None,\
+                "When use optimize or _save_jit is called for the first time, it must have and opendr Image input"
+            if not isinstance(img, Image):
+                img = Image(img)
+            img = img.opencv()
+            if not self.dummy_input:
+                self.dummy_input = self.predictor.preprocessing(img)
+
+        torch.onnx.export(
+            self.predictor,
+            self.dummy_input,
+            export_path,
+            verbose=verbose,
+            keep_initializers_as_inputs=True,
+            do_constant_folding=do_constant_folding,
+            opset_version=11,
+            input_names=['data'],
+            output_names=['output'],
+        )
+
+        metadata = {"model_paths": ["nanodet_{}.onnx".format(self.cfg.check_point_name)], "framework": "pytorch",
+                    "format": "onnx", "has_data": False, "inference_params": {}, "optimized": True,
+                    "optimizer_info": {}, "classes": self.classes}
+
+        with open(os.path.join(onnx_path, "nanodet_{}.json".format(self.cfg.check_point_name)),
+                  'w', encoding='utf-8') as f:
+            json.dump(metadata, f, ensure_ascii=False, indent=4)
+
+        if verbose:
+            print("finished exporting onxx")
+            self.logger.log("finished exporting onxx")
+
+        try:
+            import onnxsim
+            import onnx
+            if verbose:
+                print("start simplifying onnx")
+                self.logger.log("start simplifying onnx")
+            input_data = {"data": self.dummy_input.detach().cpu().numpy()}
+            model_sim, flag = onnxsim.simplify(export_path, input_data=input_data)
+            if flag:
+                onnx.save(model_sim, export_path)
+                if verbose:
+                    self.logger.log("simplify onnx successfully")
+            else:
+                if verbose:
+                    self.logger.log("simplify onnx failed")
+        except:
+            print("For compression in optimized models, install the onnxsim dependencies and rerun optimize")
```

Comment:
Looks like the try/except clause refers to whether the dependencies are installed, but covers a whole lot more. It would be better to use it around only the imports and use the specific exception that will occur if the dependencies are not installed.